### ISSUE TITLE
ddclient.conf copied with bad permissions

### ISSUE THREAD

**Issue by mtalexan** [linuxserver/docker-ddclient#32]
The startup script within the container at `/etc/cont-init.d/30-config` copies the ddclient.conf file from `/config/ddclient.conf` to `/ddclient.conf`.  It then executes the `chown` and `chmod` commands, which do not affect the `/ddclient.conf` file.  When the ddclient is started, it is directed to use the `/ddclient.conf` for the configuration file, and reports it is unable to open the file due to permissions errors.
Since the permissions changes are not applied to the copied file, but are applied to the `/config` folder after the contents are copied out of it, the permissions of the files on first-time start of the container are what define the permissions on the `/ddclient.conf` file.  Also, stopping and restarting the container will have different results since the `/config/ddclient.conf` file had the permissions changed during boot.

I believe the intended code in `/etc/cont-init.d/30-config` of the container should be:
```
#!/usr/bin/with-contenv bash

# make our folders
mkdir -p \
        /var/cache/ddclient \
        /var/run/ddclient

# copy default config if not present in /config
[[ ! -e /config/ddclient.conf ]] && \
        cp /defaults/ddclient.conf /config

# copy config from /config to root
cp /config/ddclient.conf /ddclient.conf

# permissions
chown -R abc:abc \
        /ddclient.conf \
        /var/cache/ddclient \
        /var/run/ddclient
chmod -R 600 \
        /ddclient.conf
```
Basically the `chown` and `chmod` should act on `/ddclient.conf` instead of `/config`.
------------------------------

## Expected Behavior

Starting the container with a proven valid ddclient.conf file in the /config folder should use that config.

## Current Behavior

ddclient within the container throws an error about incorrect permissions and doesn't function

## Steps to Reproduce

1. Copy the provided docker-compose
2. Create a new directory with UID and GID matching that used in the docker-compose
3. Put any ddclient.conf file in the directory (can be blank)
4. chown the ddclient.conf file to match the UID and GID from the docker-compose
5. chmod the ddclient.conf file to 600
6. Start the docker container
7. Look at the docker logs and notice the permission denied errors on the /ddclient.conf file

## Environment
**OS:**  Ubuntu 18.04.3
**CPU architecture:** x86_64
**How docker service was installed:**      

```
sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
sudo apt-get update
apt-cache policy docker-ce
sudo apt-get install docker-ce
```

## Command used to create docker container (run/create/compose/screenshot)

```
services:
  ddclient:
    image: linuxserver/ddclient
    container_name: ddclient
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/Los_Angeles
    volumes:
      - /docker/ddclient/config:/config
    restart: unless-stopped
```
```
$ sudo tree -pug /docker/ddclient                                                                                                                          
/docker/ddclient
├── [drw------- docker     docker    ]  config
│   └── [-rw------- docker     docker    ]  ddclient.conf
└── [-rwxr-xr-- docker     docker    ]  docker-compose.yml

1 directory, 2 files

$ id docker
uid=1000(docker) gid=1000(docker) ...
```

## Docker logs

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing... 
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 10-adduser: executing... 

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \ 
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
We gratefully accept donations at:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 30-config: executing... 
[cont-init.d] 30-config: exited 0.
[cont-init.d] 99-custom-files: executing... 
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
Setting up watches.
Watches established.
[services.d] done.
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
stat() on closed filehandle FD at /usr/bin/ddclient line 1117.
Use of uninitialized value $mode in bitwise and (&) at /usr/bin/ddclient line 1118.
readline() on closed filehandle FD at /usr/bin/ddclient line 1130.
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
WARNING:  file /ddclient.conf: Cannot open file '/ddclient.conf'. (Permission denied)
```

**Comment by davidosomething** [comment]
this appears fixed by #31 which copies the file over, but the root cause (willy nilly copying files to root path) is still concerning

**Comment by aptalca** [comment]
PR welcome
